includes/class-block-patterns.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * The Block Patterns post type.
+ *
+ * @package CoBlocks
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * The Block Patterns post type.
+ *
+ * @since NEXT
+ */
+class CoBlocks_Block_Patterns {
+	/**
+	 * This plugin's instance.
+	 *
+	 * @var CoBlocks_Block_Patterns
+	 */
+	private static $instance;
+
+	/**
+	 * Registers the plugin.
+	 *
+	 * @return CoBlocks_Block_Patterns
+	 */
+	public static function register() {
+		if ( null === self::$instance ) {
+			self::$instance = new CoBlocks_Block_Patterns();
```

Comment:
I feel like this is prone to error doing copy paste. I would suggest using `new self()` instead. Also since we are over php 5.4 now I feel like we could introduce a singleton trait that we can reuse everywhere.

```php
trait Singleton {
	/**
	 * The plugin instance.
	 *
	 * @var Plugin
	 */
	private static $instance = null;
	/**
	 * Return the plugin instance.
	 *
	 * @return Plugin
	 */
	public static function load() {
		if ( ! self::$instance ) {
			self::$instance = new self();
		}
		return self::$instance;
	}
	/**
	 * Reset the plugin instance.
	 */
	public static function reset() {
		self::$instance = null;
	}
}
```

Reply:
Good call. This will be added in the next push.